nodes/project-link.js

```diff
@@ -497,7 +497,12 @@ module.exports = function (RED) {
                     options = Object.assign({}, defaultOptions, options)
 
                     if (RED.settings.flowforge.projectLink.broker.clientId) {
-                        options.clientId = RED.settings.flowforge.projectLink.broker.clientId + ':n'
+                        options.clientId = RED.settings.flowforge.projectLink.broker.clientId
+                    } else {
+                        // If no clientId specified, use 'username' with ':n' appended.
+                        // This ensures uniqueness between this and the launcher/device's own
+                        // client connection.
+                        options.clientId = RED.settings.flowforge.projectLink.broker.username + ':n'
```

Comment:
What happens if username is blank? Wont all clients will get clientId `':n`

Reply:
There is no valid configuration when username can be blank. Broker url/username/password are all required.